### ISSUE TITLE
ParSys Sprint G Goal

### ISSUE THREAD

**Issue by hmitsch** [mozilla/parsys#253]
# Sprint Goal
Following the _Sprint F Retrospective_, we experiment with a more Kanban-oriented way of mapping out the next 2 weeks. This is the prioritization we landed on at the end of our Sprint Planning meeting.
![image.png](https://images.zenhubusercontent.com/57641233bf93871ccd9cc8fe/bc3a1f4c-b47f-4479-bb71-3152acafb5e2)

# Preparation for the planning meeting (aka former Sprint Goal)

## Be a good Open Innovation citizen
* ~(IAM) Common Voice: Integration support~
* ~(Infrastructure) Support SuMo Strategic Research: Hand off secure and redacted Kitsune DB for Tableau analysis~
* (OI Products) FOSS Heartbeat: Make devtools repo parseable (multi-file support) and clarify next steps
* (OI Products) Explore SUMO Playstore tool: Can we build something quick, scrappy, non-sustainable with effort measured in days, not weeks/months?
* (Contributor PRs) Keep the pipe clean, review & merge in a timely manner
* (OI Products) Coral Meeting

## (IAM) Firefox Accounts in Mozilla IAM
* Prepare launch comms
* enable fxa on mozillians staging for login and account verification 
   * Mozillians shows "none" as login method: adding new IDPs is fairly easy, its a dictionary with security levels. let's tackle it in the next sprint

## (IAM) OIDC Django Library
* Support XHR requests in the Token Refresh middleware
* Support extra_params like connection type in Authentication Request class

## (OI Products) Blind Code Review Pilot
* Submit Q2 project charter
* tl;dr newsletter posting
* Transition repo to a mozilla/parsys Github org

## (CoSS) Launch a Pilot Event on Splash
* More Splash Salesforce Integration testing
* Configure https://events.mozilla.org
* More acceptance criteria TBD

## (IAM) Stop starting, and start finishing / Operational improvements
* Team Drives are renamed in PROD
* ~Move Auth0 & CIS tests into new Jenkins environment~
* ~Add 2 access groups tests for Mozillians.org~
* ~(SSO) Dashboard: Retrieve Avatar URLs from CIS~
* Mozillians-Salesforce integration (Basket): Country update (docs) and end-to-end validation
* "Tech 1337" name is shown for all users in sso prod [#258](https://github.com/mozilla-iam/sso-dashboard/issues/258)
* Improve error page wayfinding [#205](https://github.com/mozilla-iam/sso-dashboard/issues/205)
* Bug: [Unclear email about NDA](https://bugzilla.mozilla.org/show_bug.cgi?id=1450501)
* Bug: [Open Sans font family in mozillians.org does not include non-English characters](https://bugzilla.mozilla.org/show_bug.cgi?id=779948)

## (IAM) LDAP publisher / Auth0 publisher
* (Staging-only) LDAP groups are exposed via the Person API
* (Staging-only) auth0 creates new users in Mozilla IAM
Info: this enables "staff data available in Mozillians"

## (IAM) Change User Profile Schema (Design)
* Discuss changes in Tech Alignment, map out work for next sprint
* Discuss architecture around future public API (Person API)
   * Example: github ID is visible in the user profile https://github.com/mozilla-iam/mozilla-iam/issues/34
* Review discussion outcomes with Jonas from Taskcluster

## (IAM) Make Kubernetes Infrastructure real
* ~Tech Alignment (or separate meeting): What's the CI/CD process and release process?~
* Moderator is migrated

## [XL] (IAM) Mozillians.org Repositioning: UX Research
* [Plan](https://docs.google.com/document/d/1km2cG-bW1Z_NUclJuZuzJCw6qMi1Rcg1FjYkD6nJHL0/edit)
* Acceptance criteria TBD

## NLX.next
*  logging in to github with an LDAP email = endless loop [#64](https://github.com/mozilla-iam/auth0-custom-lock/issues/64)
* Re-try to use prompt param for autologin [#91](https://github.com/mozilla-iam/auth0-custom-lock/issues/91)
* NLX autologin causes Auth0 /logout to not log the user out if the user is using the Google or GitHub IDPs [#96](https://github.com/mozilla-iam/auth0-custom-lock/issues/96)
*  During NLX autologin an unecesary call to Auth0 /connections endpoint is made [#97](https://github.com/mozilla-iam/auth0-custom-lock/issues/97)
* Store user name in localstorage [#9](https://github.com/mozilla-iam/auth0-custom-lock/issues/9)
* Q2 roadmap for NLX (including [IAM User Stories around NLX, Identity Selection and Login Preferences](https://docs.google.com/spreadsheets/d/1tTLpRBzS8DkS_AA6_-u_89xFPmtAz6AF2HRQqXMyR3k/edit#gid=0) and [NLX known issues](https://docs.google.com/document/d/1tLgevSeGzosgvV4k-KayJBoMMAxrGdkYWP0xPpd4mVs/edit))
* Trying to login to sso dashboard after logout shows first the autologin message, then nlx [#62](https://github.com/mozilla-iam/auth0-custom-lock/issues/62)
* No option for two login methods (ldap vs vouched, non-mozillians) [#43] (https://github.com/mozilla-iam/auth0-custom-lock/issues/43) 
* After linking a new identity in mozillians, nlx tries to do the autologin with that identity [#249](https://github.com/mozilla/parsys/issues/249)
* https://discourse.mozilla.org/t/sso-password-field-not-recognized-by-dashlane/26008/1
* https://discourse.mozilla.org/t/repeated-login-prompts-in-outlook/25921

## Upgrade Project Charters for April Operations Review
* CoSS
* ~Blind Code Reviews~
* IAM

## Discourse
* Do we need more test automation? Can we continue to "just press the PROD deploy button" whenever we feel like it? (Ask Viorela and Leo)

# Backlogged

## [L] (IAM) Resilient Operations: Tracing in the IAM Stack
* Logs from Mozillians.org and CIS are in the same place

## [L] (IAM) Individual group memberships expiry dates
* UX design for setting the expiry date
* As an access group curator, I can set an individual expiry date

## (IAM) NLX Usertesting.com Analysis
* Insights are captured in Github issues, estimated and prioritized

## (IAM) NLX deployment automation
* NLX can be deployed without manual intervention

## GCP Reference Arch for Federated Access
* Currently blocked: Credentials missing (GCP account creation)
* Meeting with Google engineers next week

* (Operations) Runbook: its complete for Mozillians.org, still requires copy & paste for the other ParSys properties
* OSSN: Container hosting for GSoC
* Discourse: As a Staff member I want to login using LDAP and get email notifications on a separate email address in order to separate inboxes for paid work and community work.
